server/StudySharp.API/Controllers/CourseController.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System.Threading.Tasks;
+using AutoMapper;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using StudySharp.API.Requests.Courses;
+using StudySharp.API.Responses.Courses;
+using StudySharp.ApplicationServices.Commands;
+using StudySharp.ApplicationServices.Queries;
+using StudySharp.Domain.General;
+
+namespace StudySharp.API.Controllers
+{
+    [Authorize]
+    [ApiController]
+    [Route("api/[controller]")]
+
+    public class CourseController : ControllerBase
+    {
+        private readonly IMediator _mediator;
+        private readonly IMapper _mapper;
+
+        public CourseController(IMediator mediator, IMapper mapper)
+        {
+            _mediator = mediator;
+            _mapper = mapper;
+        }
+
+        [HttpPost("add")]
+        public async Task<OperationResult> Add([FromBody] AddCourseRequest addCourseRequest)
+        {
+            var addCourseCommand = _mapper.Map<AddCourseCommand>(addCourseRequest);
+            return await _mediator.Send(addCourseCommand);
+        }
+
+        [HttpPost("remove")]
+        public async Task<OperationResult> Remove([FromBody] RemoveCourseByIdRequest removeCourseByIdRequest)
```

Comment:
`[HttpDelete("{id:int}")]`

And add `[BindProperty(Name = "id", SupportsGet = true)]` to RemoveCourseByIdRequest model like 
![image](https://user-images.githubusercontent.com/32524389/134772045-71d310a6-400f-4de2-a547-8886cc66a925.png)
This is necessary in order for the id in the route parameter to be linked to the model

And rewrite [FromBody] to [FromRoute]

[FromBody] - request body in json like 
```
{
"name":"Anna",
"surname":"Conor"
}
```
About difference [FromRoute] and [FromQuery] read [here](https://stackoverflow.com/questions/47976168/are-fromuri-and-fromquery-the-same/48000116)

Reply:
And remember, we have 4 major types of HTTP request:
HttpPost - for add models, 
HttpGet - get model or list of models
HttpPut - update models
HttpDelete - remove/delete models